MvvmCross/Binding/iOS/Views/MvxCollectionViewSource.cs

```diff
@@ -80,14 +81,45 @@ protected override object GetItemAt(NSIndexPath indexPath)
         {
             return this.ItemsSource?.ElementAt(indexPath.Row);
         }
-
-        private void CollectionChangedOnCollectionChanged(object sender,
-                                                          NotifyCollectionChangedEventArgs
-                                                              notifyCollectionChangedEventArgs)
+            
+        /// <summary>
+        /// Wait for all animations to finish
+        /// </summary>
+        public async Task WaitAnimationsCompleted()
         {
-            this.ReloadData();
+            await CollectionView.PerformBatchUpdatesAsync(() => { }); 
         }
 
+        protected virtual void CollectionChangedOnCollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
+        {
+            if (args.Action == NotifyCollectionChangedAction.Remove)
+            {
+                CollectionView.PerformBatchUpdates(() => 
+                {
+                    var startIndex = args.OldStartingIndex;
+                    var indexes = new NSIndexPath[args.OldItems.Count];
+                    for(var i=0; i<indexes.Length; i++)
+                        indexes[i] = NSIndexPath.FromRowSection(startIndex+i, 0);
+                    CollectionView.DeleteItems(indexes);
+                }, ok => { });
+            }
+            else if (args.Action == NotifyCollectionChangedAction.Add)
+            {
+                CollectionView.PerformBatchUpdates(() =>
+                {
+                    var startIndex = args.NewStartingIndex;
+                    var indexes = new NSIndexPath[args.NewItems.Count];
+                    for (var i = 0; i < indexes.Length; i++)
+                        indexes[i] = NSIndexPath.FromRowSection(startIndex + i, 0);
+                    CollectionView.InsertItems(indexes);
+                }, ok => { });
+            }
+            else
```

Comment:
Already supported or not needed

Benjamin

Le samedi 4 juin 2016, Tomasz Cielecki <notifications@github.com> a écrit :

> In MvvmCross/Binding/iOS/Views/MvxCollectionViewSource.cs
> <https://github.com/MvvmCross/MvvmCross/pull/1333#discussion_r65807543>:
>
> > +                        indexes[i] = NSIndexPath.FromRowSection(startIndex+i, 0);
> > +                    CollectionView.DeleteItems(indexes);
> > +                }, ok => { });
> > +            }
> > +            else if (args.Action == NotifyCollectionChangedAction.Add)
> > +            {
> > +                CollectionView.PerformBatchUpdates(() =>
> > +                {
> > +                    var startIndex = args.NewStartingIndex;
> > +                    var indexes = new NSIndexPath[args.NewItems.Count];
> > +                    for (var i = 0; i < indexes.Length; i++)
> > +                        indexes[i] = NSIndexPath.FromRowSection(startIndex + i, 0);
> > +                    CollectionView.InsertItems(indexes);
> > +                }, ok => { });
> > +            }
> > +            else
>
> What about MoveItem?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MvvmCross/MvvmCross/pull/1333/files/3776e268fee7e87017eabd9d39113d4735b71a44#r65807543>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALpJCawSlzelNCPCWEyNyAEMRogGE-xks5qIeXIgaJpZM4Ikd-V>
> .
>